cpp/Esri/ArcGISRuntime/Toolkit/CoordinateConversionOption.h

```diff
@@ -0,0 +1,247 @@
+  Q_PROPERTY(CoordinateType outputMode READ outputMode WRITE setOutputMode NOTIFY outputModeChanged)
+  Q_PROPERTY(bool addSpaces READ addSpaces WRITE setAddSpaces NOTIFY addSpacesChanged)
+  Q_PROPERTY(int precision READ precision WRITE setPrecision NOTIFY precisionChanged)
+  Q_PROPERTY(int decimalPlaces READ decimalPlaces WRITE setDecimalPlaces NOTIFY decimalPlacesChanged)
+  Q_PROPERTY(MgrsConversionMode mgrsConversionMode READ mgrsConversionMode WRITE setMgrsConversionMode NOTIFY mgrsConversionModeChanged)
+  Q_PROPERTY(LatitudeLongitudeFormat latLonFormat READ latLonFormat WRITE setLatLonFormat NOTIFY latLonFormatChanged)
+  Q_PROPERTY(UtmConversionMode utmConversionMode READ utmConversionMode WRITE setUtmConversionMode NOTIFY utmConversionModeChanged)
+public:
+  /*! Fundamental coordinate format. */
+  enum CoordinateType
+  {
+    Gars,
+    GeoRef,
+    LatLon,
+    Mgrs,
+    Usng,
+    Utm
+  };
+  Q_ENUM(CoordinateType)
+
+public:
+  /*!
+   * \brief Constructor
+   * \param parent Owning parent QObject.
+   */
+  Q_INVOKABLE CoordinateConversionOption(QObject* parent = nullptr);
+
+  /*!
+   * \brief Destructor
+   */
+  ~CoordinateConversionOption() override;
+
+  /*!
+   * \brief Converts a Point to a QString based on the properties set in this
+   * CoordinateConversionOption
+   * 
+   * \param point Point to convert to string.
+   * \return String representation of point with the current formatting options
+   * applied.
+   */
+  Q_INVOKABLE QString prettyPrint(const Point& point) const;
+
+  /*!
+   * \brief Given a string called point, converts it to a Point using the
+   * current formatting options applied.
+   * 
+   * This conversion has the chance of failing, in which case an invalid
+   * default-constructed Point will be returned.
+   * 
+   * \param point QString to conver to a Point.
+   * \param spatialReference SpatialReference to use in conversion.
+   * \return QString as a Point. 
+   */
+  Q_INVOKABLE Point pointFromString(const QString& point,
+                                    const SpatialReference& spatialReference);
+
+  /*!
+   * \brief The user friendly name of this option.
+   */
+  QString name() const;
+
+  /*!
+   * \brief Set name.
+   * \sa name
+   */
+  void setName(const QString& name);
+
+  /*!
+   * \brief The conversion type of this option.
+   * \return CoordinateType 
+   */
+  CoordinateType outputMode() const;
+
+  /*!
+   * \brief Set converion type.
+   * \sa outputMode
+   */
+  void setOutputMode(CoordinateType outputMode);
+
+  /*!
+   * \brief Flags whether to add spaces in notation (if applicable).
+   */
+  bool addSpaces() const;
+
+  /*!
+   * \brief Set addSpaces.
+   * \sa addSpaces
+   */
+  void setAddSpaces(bool addSpaces);
+
+  /*!
+   * \brief Precision of conversion format (if applicable).
+   */
+  int precision() const;
+
+  /*!
+   * \brief Set the precision.
+   * \sa precision
+   */
+  void setPrecision(int precision);
```

Comment:
i wonder if decimal places and precision should both be unsigned

Reply:
This is a good point, but the `CoordinateFormatter` accepts only ints, so will stick to that.